test/test_uncertainty.py

```diff
@@ -0,0 +1,139 @@
+head_configs = {
+    "umal": {
+        "head": "umal",
+        "loss": "UMALLoss",
+        "n_taus": 32,
+        "umal_extend_batch": True,
+        "tau_down": 0.1,
+        "tau_up": 0.9,
+    },
+    "cmal": {
+        "head": "cmal",
+        "loss": "CMALLoss",
+        "n_distributions": 3,
+    },
+    "gmm": {
+        "head": "gmm",
+        "loss": "GMMLoss",
+        "n_distributions": 3,
+    }
+}
+
+def build_full_config(head):
+    """Builds a full config dictionary for the given head."""
+    config = head_configs[head].copy()
+
+    # Only add common uncertainty fields if the head supports them
+    if head in ["umal", "cmal", "gmm"]:
+        config.update(common_uncertainty_config)
+    
+    return config
+
+
+@pytest.mark.parametrize("mc_dropout", [False, True])
+@pytest.mark.parametrize("negative_sample_handling", ["none", "clip", "truncate"])
+@pytest.mark.parametrize("head", ["umal", "cmal", "gmm"])
+def test_daily_uncertainty(get_config: Fixture[Callable[[str], dict]],
+                           daily_dataset: Fixture[str],
+                           single_timescale_forcings: Fixture[str],
+                           head: str,
+                           mc_dropout: bool,
+                           negative_sample_handling: str):
+    """Test uncertainty output for different heads, losses, and negative sample handling strategies."""
+    
+    config = get_config('daily_uncertainty')  # Load a generic daily config
+
+    basin = '01022500'
+
+    # Dynamically build the basic config
+    update_dict = {
+        'head': head,
+        'dataset': daily_dataset['dataset'],
+        'data_dir': config.data_dir / daily_dataset['dataset'],
+        'negative_sample_handling': negative_sample_handling,
+        'mc_dropout': mc_dropout,
+        'target_variables': daily_dataset['target'],
+        'forcings': single_timescale_forcings['forcings'],
+        'dynamic_inputs': single_timescale_forcings['variables'],
+    }
+
+    if negative_sample_handling == 'truncate':
+        update_dict['negative_sample_max_retries'] = 3
+
+    config.update_config(update_dict)
+
+    # Merge in the head-specific parameters dynamically
+    head_specific_config = build_full_config(head)
+    config.update_config(head_specific_config)
+
+    # Start training and evaluation
+    print(f"\n[TEST] head={head}, loss={config.head}, neg_sample_handling={negative_sample_handling}")
+    start_training(config)
+    start_evaluation(cfg=config, run_dir=config.run_dir, epoch=1, period='test')
+
+    # Sanity check of uncertainty outputs
+    _check_uncertainty_output(config, basin, negative_sample_handling)
+
+
+def _check_uncertainty_output(config: Config, basin: str, negative_sample_handling: str):
+    """Perform basic sanity checks of uncertainty predictions.
+
+    Checks that:
+        -the results file has the correct date range, 
+        -the observed discharge in the file is correct, 
```

Comment:
I think this check is missing. Should be easy to copy from the existing test though.